File: src/main/java/com/tibiawiki/domain/utils/PropertiesUtil.java
package com.tibiawiki.domain.utils;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.InputStream;
import java.util.Properties;

public class PropertiesUtil {

    private static final Logger LOG = LoggerFactory.getLogger(PropertiesUtil.class);

    private PropertiesUtil() {
        // don't instantiate this class, it has only static members
    }

    public static String getUsername() {
        return getProperty("username");
    }

    public static String getPassword() {
        return getProperty("password");
    }

    private static String getProperty(String propertyName) {
        try {
            Properties props = new Properties();
            ClassLoader classloader = Thread.currentThread().getContextClassLoader();
            InputStream is = classloader.getResourceAsStream("credentials.properties");

            if (is != null) {
                props.load(is);
                return props.getProperty(propertyName);
            } else {
                LOG.warn("Could not read requested propertyName {} from file 'credentials.properties'.", propertyName);
                return null;
            }
        } catch (Exception ex) {
            LOG.error(String.valueOf(ex));
        }
        return null;
    }
}

File: src/main/kotlin/com/tibiawiki/domain/utils/PropertiesUtil.kt
package com.tibiawiki.domain.utils

import org.slf4j.LoggerFactory

object PropertiesUtil {
    private val LOG = LoggerFactory.getLogger(PropertiesUtil::class.java)

    fun getUsername(): String? = getProperty("username")

    fun getPassword(): String? = getProperty("password")

    private fun getProperty(propertyName: String): String? {
        return try {
            val props = java.util.Properties()
            val classloader = Thread.currentThread().contextClassLoader
            val inputStream = classloader.getResourceAsStream("credentials.properties")
            if (inputStream != null) {
                inputStream.use { props.load(it) }
                props.getProperty(propertyName)
            } else {
                LOG.warn("Could not read requested propertyName {} from file 'credentials.properties'.", propertyName)
                null
            }
        } catch (ex: Exception) {
            LOG.error(ex.toString())
            null
        }
    }
}
